torch/csrc/distributed/c10d/comm.cpp

```diff
@@ -31,7 +31,7 @@ class BroadcastWork {
         flat_tensor_.front(), bucket_tensors_);
     TORCH_INTERNAL_ASSERT(output_tensors.size() == bucket_tensors_.size());
     for(const auto i : c10::irange(output_tensors.size())) {
-      bucket_tensors_[i].copy_(output_tensors[i], /*non_blocking=*/true);
+      bucket_tensors_[i] = output_tensors[i];
```

Comment:
I originally had one concern around change of semantics here where output_tensors could be modified which would also affect bucket_tensors (which wouldn't happen before this PR). However, after looking closely at how output_tensors are built, I realized this:

1. In the case where we have multiple bucket tensors, flat_tensor_ is actually a brand new Tensor and is only used in this class. `output_tensors` are just views of this flat_tensor_ and hence there is no harm in just assigning like this.
2. In case there is just a single bucket tensor, flat_tensor_ is basically the same as `bucket_tensor[0]` and the assignment is harmless.

Although, this copy was introduced in https://github.com/pytorch/pytorch/pull/20234, so I'll let @mrshenli double check this and stamp it.

Reply:
hmm, I might miss sth. I was assuming this change would make `bucket_tensors_[i]` point to the new tensor/view `output_tensors[i]` instead of updating the content in the input tensors. So, this might behave differently than before if users have created views from the input tensor (i.e., `bucket_tensors `) before calling this API?